repair/repair.cc

```diff
@@ -959,9 +959,22 @@ static future<> repair_cf_range(repair_info& ri,
 // Comparable to RepairSession in Origin
 static future<> repair_range(repair_info& ri, const dht::token_range& range) {
     auto id = utils::UUID_gen::get_time_UUID();
-    return do_with(get_neighbors(ri.db.local(), ri.keyspace, range, ri.data_centers, ri.hosts), [&ri, range, id] (std::vector<gms::inet_address>& neighbors) {
+    repair_neighbors neighbors = ri.neighbors.empty() ? repair_neighbors(get_neighbors(ri.db.local(), ri.keyspace, range, ri.data_centers, ri.hosts)) : ri.neighbors[range];
```

Comment:
It is better to always have the caller prepare ri.neighbors(). Functions that behave differently depending on contents of parameters are tricky to understand.

Reply:
I want to avoid store the neighbors if we can simply calculate it on the fly. How about adding a helper ri.get_repair_neighbors(range) which hides all the details from the user.

Reply:
That works well.

Reply:
Fixed.